sync/src/tasks/mock.rs

```diff
@@ -278,6 +278,12 @@ impl SyncNodeMocker {
         self.chain_mocker.head()
     }
 
+    pub fn dag_fork_number(&self) -> Result<BlockNumber> {
```

Comment:
For non builtin network, this function will fail? It's better to support user-defined network.

Reply:
Yes, it will fail. 

No test case uses customized network BuiltinNetworkID for now. 

Because the customized network has user-defined network id, it cannot go through the built-in-network logic successfully. 

The customized network test case must be written case by case, you may refer to test_init_script or other cases that use ChainNetwork::new_custom to create a customized network for running each test.

If you insist, tell me which case has to be running in customized network.

Reply:
Check the starcoin cookbook please. It's merely a suggestion, do what you like.